src/mmr.rs

```diff
@@ -271,37 +270,22 @@ impl<T: Clone + PartialEq, M: Merge<Item = T>, S: MMRStoreReadOps<T>> MMR<T, M,
                 }
             };
 
-            let queue_front_pos = queue.front().map(|(pos, _)| pos);
-            if Some(&sib_pos) == queue_front_pos {
+            if Some(&sib_pos) == queue.front().map(|(_, pos)| pos) {
                 // drop sibling
                 queue.pop_front();
-            } else if queue_front_pos.is_none()
-                || !is_descendant_pos(
```

Comment:
I think that's a first time I see this code, so DQ: we don't need that `is_descendant_pos` call anymore? I guess it has something to do with this new ordering, but `is_descendant_pos` is not a trivial comparison, so just want to check

Reply:
Yes, because now we start building the proof from the lower heights, and we advance incrementally toward the higher ones, so we shouldn't stumble upon descendants anymore

Reply:
> Yes, because now we start building the proof from the lower heights, and we advance incrementally toward the higher ones, so we shouldn't stumble upon descendants anymore

exactly: a parent's position is always strictly greater than its childrens' or any other descendants, since it can only be created after all its descendants exist (nit: it's a confusing pity that our terminology is the complete reverse of biology 🙉).
So having the queue's items ordered by increasing position means any node's ancestors are always processed after said node - hence we no longer have to perform this expensive check.

> but is_descendant_pos is not a trivial comparison, so just want to check